test/datatable.jl

```diff
@@ -342,6 +342,40 @@ module TestDataTable
     @test find(c -> isa(c, NullableCategoricalArray), categorical!(DataTable(A=1:3, B=4:6), [1]).columns) == [1]
     @test find(c -> isa(c, NullableCategoricalArray), categorical!(DataTable(A=1:3, B=4:6), 1).columns) == [1]
 
+    @testset "unstack nullable promotion" begin
+        long = DataTable(Any[repeat([:c, :d], inner=12),
+                             vcat(linspace(0, 1, 12), linspace(2, 1, 12)),
```

Comment:
Can't you find a simpler example which doesn't involve `repeat` nor `linspace`? Since the goal here is to test promotion, focus on testing types rather than contents. BTW, types are not tested.